packages/anoncreds/src/services/AnonCredsHolderService.ts

```diff
@@ -19,8 +21,10 @@ export interface AnonCredsHolderService {
     metadata?: Record<string, unknown>
   ): Promise<string>
 
-  // TODO: indy has different return types for the credential
-  getCredential(agentContext: AgentContext, options: GetCredentialOptions): Promise<CredentialInfo>
+  // TODO: this doesn't actually return the credential, as the indy-sdk doesn't support that
+  // We could come up with a hack (as we've received the credential at one point), but for
+  // now I think it's not that much of an issue
+  getCredential(agentContext: AgentContext, options: GetCredentialOptions): Promise<AnonCredsCredentialInfo>
 
   createCredentialRequest(
```

Comment:
What about MasterSecret creation and storage? It is part of AnonCredsApi, isn't it? If that's the case, will it be passed in `CreateCredentialRequestOptions` and `CreateProofOptions`?

Reply:
Yeah it should be handled by the AnonCreds Api / services (prob the holder service), and removed from the wallet interface.

Reply:
I figured out that it's only needed for CreateCredentialRequestOptions and can even be optional, using a default one as in 99% of the cases it will be a single master secret. I've implemented some draft classes in my PR https://github.com/TimoGlastra/aries-framework-javascript/pull/18